pkg/cloudprovider/cloudprovider.go

```diff
@@ -234,16 +238,24 @@ func (c *CloudProvider) GetSupportedNodeClasses() []status.Object {
 	return []status.Object{&v1.EC2NodeClass{}}
 }
 
-func getTags(ctx context.Context, nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.NodeClaim) map[string]string {
+func getTags(ctx context.Context, nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.NodeClaim) (map[string]string, error) {
+	if _, found := lo.Find(v1.RestrictedTagPatterns, func(exp *regexp.Regexp) bool {
+		for key := range nodeClass.Spec.Tags {
+			if exp.MatchString(key) {
+				return true
+			}
+		}
+		return false
+	}); found {
+		return nil, fmt.Errorf("tag validation bypassed")
```

Comment:
Should we be more specific in our error messaging -- this is basically going to print the same error twice due to the wrapping up above